src/utils/credentials-helper.js

```diff
@@ -0,0 +1,244 @@
+import {
+  getSolidDataset,
+  getThing,
+  createSolidDataset,
+  buildThing,
+  createThing,
+  saveSolidDatasetAt,
+  saveSolidDatasetInContainer,
+  setThing,
+  getStringNoLocale
+} from '@inrupt/solid-client';
+import { Session } from '@inrupt/solid-client-authn-browser';
+
+import { createDocAclForUser } from './session-helper';
+
+import { RDF_PREDICATES } from '../constants';
+
+const generateRsaKeyStrings = async () => {
+  const { privateKey, publicKey } = await window.crypto.subtle.generateKey(
```

Comment:
The `WebCrypto` API adds a cryptography library usable in the browser:
https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API